plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -269,54 +277,54 @@ class AzBatchService implements Closeable {
+        if( !config.batch().accountKey )
+            throw new IllegalArgumentException("Missing Azure Batch account key -- Specify it in the nextflow.config file using the setting 'azure.batch.accountKey'")
 
-        }
+        return new AzureNamedKeyCredential(config.batch().accountName, config.batch().accountKey)
     }
 
-    protected createBatchCredentialsWithServicePrincipal() {
+    protected TokenCredential createBatchCredentialsWithServicePrincipal() {
         log.debug "[AZURE BATCH] Creating Azure Batch client using Service Principal credentials"
 
-        final batchEndpoint = "https://batch.core.windows.net/";
-        final authenticationEndpoint = "https://login.microsoftonline.com/";
+        return new ClientSecretCredentialBuilder()
+                .tenantId(config.activeDirectory().tenantId)
+                .clientId(config.activeDirectory().servicePrincipalId)
+                .clientSecret(config.activeDirectory().servicePrincipalSecret)
+                .build()
+    }
 
-        def servicePrincipalBasedCred = new BatchApplicationTokenCredentials(
-                config.batch().endpoint,
-                config.activeDirectory().servicePrincipalId,
-                config.activeDirectory().servicePrincipalSecret,
-                config.activeDirectory().tenantId,
-                batchEndpoint,
-                authenticationEndpoint
-        )
+    protected TokenCredential createBatchCredentialsWithManagedIdentity() {
+        log.debug '[AZURE BATCH] Creating Azure Batch client using Managed Identity credentials'
 
-        return servicePrincipalBasedCred
+        return new ManagedIdentityCredentialBuilder()
+                .clientId(config.managedIdentity().clientId)
+                .build()
     }
 
     protected BatchClient createBatchClient() {
         log.debug "[AZURE BATCH] Executor options=${config.batch()}"
 
-        def cred = config.activeDirectory().isConfigured()
-                ? createBatchCredentialsWithServicePrincipal()
-                : createBatchCredentialsWithKey()
+        final builder = new BatchClientBuilder()
+        if( config.managedIdentity().isConfigured() )
+            builder.credential( createBatchCredentialsWithManagedIdentity() )
+        else if( config.activeDirectory().isConfigured() )
+            builder.credential( createBatchCredentialsWithServicePrincipal() )
+        else if( config.batch().endpoint || config.batch().accountKey || config.batch().accountName )
+            builder.credential( createBatchCredentialsWithKey() )
 
-        // Create batch client
-        def client = BatchClient.open(cred as BatchCredentials)
+        if( config.batch().endpoint )
+            builder.endpoint(config.batch().endpoint)
 
-        Global.onCleanup((it)->client.protocolLayer().restClient().close())
+        // Global.onCleanup((it)->client.protocolLayer().restClient().close())
```

Comment:
Is this cleanup hook still needed?

Reply:
I commented it as the api is not available anymore

Reply:
Let's remove it, if it's not needed anymore